src/formatDate.js

```diff
@@ -50,7 +50,39 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newSeparator = toFormat[toFormat.length - 1];
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const oldDateArr = date.split(oldSeparator);
+  const dateInfo = {};
+  const newDateArr = [];
+
+  for (let index = 0; index < oldDateArr.length; index++) {
+    const oldDateIndex = oldDateArr[index];
+
+    dateInfo[fromFormat[index]] = oldDateIndex;
+
+    const year = oldDateIndex < 30
+      ? `20${oldDateIndex}`
+      : `19${oldDateIndex}`;
+
+    if (fromFormat[index] === 'YY') {
+      dateInfo.YYYY = year;
+    }
+
+    if (fromFormat[index] === 'YYYY') {
+      dateInfo.YY = oldDateIndex.slice(-2);
+    }
+  }
+
+  for (const elem of toFormat) {
+    for (const key in dateInfo) {
+      if (elem === key) {
```

Comment:
can you do it without loop in a loop?

Reply:
I think so
I tried, but I did not succeed :(